credential-registry/server/django-icat-api/api_v2/models/Credential.py

```diff
@@ -42,6 +42,28 @@ def _cached(self, key, val):
             cache[key] = val
         return cache[key]
 
+    def get_local_name(self):
+        names = self.all_names
+        remote_name = None
+        for name in names:
+            if name.type == 'entity_name_assumed':
+                return name
+            else:
+                remote_name = name
+        return remote_name
+
+    def get_remote_name(self):
+        names = self.all_names
+        has_assumed_name = False
+        remote_name = None
+        for name in names:
+            if name.type == 'entity_name_assumed':
+                has_assumed_name = True
```

Comment:
Are we missing an assignment of some kind here? 
The function seems to return the last name in the list for which the type is not `entity_name_assumed`: what happens if all names are of type `entity_name_assumed`?